jupyter_events/logger.py

```diff
@@ -146,9 +146,13 @@ def emit(self, schema_id: str, version: int, data: dict, timestamp_override=None
         dict
             The recorded event data
         """
-        if not self.handlers or (schema_id, version) not in self.schemas:
-            # if handler isn't set up or schema is not explicitly listed,
-            # don't do anything.
+        # If no handlers are routing these events, there's no need to proceed.
+        if not self.handlers:
+            return
+
+        # If the schema hasn't been registered, raise a warning to make sure
+        # this was intended.
+        if (schema_id, version) not in self.schemas:
             warnings.warn(
```

Comment:
Since `warnings.warn` are issued [once per source location](https://docs.python.org/3.10/library/warnings.html#warning-categories), this will only warn about the first unregistered pair, but not subsequent unregistered pairs - shoot.  We may need `logging.warning()` here or track the missing schema/version pairs.  If we did the latter, we _could_ issue periodic warnings every nth occurrence.

We may also want to switch the order here since missing handlers will be the default, and it might be better to know there's a versioning issue prior to discovering there are no handlers.

Reply:
After talking with @Zsailer and running my own tests, it appears the `warnings.warn` frequency is a function of the _content_ (perhaps combined with the location), but not just the _location_ - so that's a good thing.

Reply:
Thank you for following up here, Kevin. Closing this thread as resolved.